fairlearn/reductions/exponentiated_gradient/exponentiated_gradient.py

```diff
@@ -0,0 +1,267 @@
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+
+"""
+This module implements the Lagrangian reduction of fair binary
+classification to standard binary classification.
+
+FUNCTIONS
+exponentiated_gradient -- optimize accuracy subject to fairness constraints
+"""
+
+from __future__ import print_function
+
+import numpy as np
+import pandas as pd
+import functools
+from fairlearn.reductions import moments
+from fairlearn.reductions import ReductionsEstimator
+from ._constants import _ACCURACY_MUL, _REGRET_CHECK_START_T, _REGRET_CHECK_INCREASE_T, \
+    _SHRINK_REGRET, _SHRINK_ETA, _MIN_T, _RUN_LP_STEP, _PRECISION, _INDENTATION
+from ._lagrangian import _Lagrangian
+from collections import namedtuple
+
+print = functools.partial(print, flush=True)
+
+
+def _mean_pred(X, hs, weights):
+    """Return a weighted average of predictions produced by classifiers in hs"""
+    pred = pd.DataFrame()
+    for t in range(len(hs)):
+        pred[t] = hs[t](X)
+    return pred[weights.index].dot(weights)
+
+
+# TODO: decide whether we want a structured object instead of the namedtuple
```

Comment:
Do we prefer this over a real object like ExponentiatedGradientResult right below?

Reply:
I thought we were preferring real objects. We're not really typed, but if we declare our return objects we have more of a chance of tracking them down.

Reply:
Cool, so you prefer the class over the namedtuple?

Reply:
That's what I've been doing (e.g. for GridSearchResult and now MetricsResult), but.... I suffer from heresies of types and compilation.